src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/config/Config.java

```diff
@@ -248,19 +275,68 @@ public Config(IGerritHudsonTriggerConfig config) {
-                categories.add(new VerdictCategory(cat.getVerdictValue(), cat.getVerdictDescription()));
-            }
-        }
+
+        assertDefaultCategories();
+
         if (config.getReplicationConfig() != null) {
             replicationConfig = new ReplicationConfig(config.getReplicationConfig());
         }
         watchdogTimeoutMinutes = config.getWatchdogTimeoutMinutes();
         watchTimeExceptionData = addWatchTimeExceptionData(config.getExceptionData());
     }
 
+    public static Integer getIntegerFromString(String str) {
+        if (str != null && !str.equals(""))
+            return Integer.parseInt(str);
+        return null;
+    }
+
+    // make sure that there is a CR and VER category
+    private void assertDefaultCategories() {
+        VerdictCategory[] defaultCategories = new VerdictCategory[]{
+            new VerdictCategory("Code-Review", "Code-Review",
+                getGerritBuildStartedCodeReviewValue() != null ? getGerritBuildStartedCodeReviewValue() : DEFAULT_GERRIT_BUILD_STARTED_CODE_REVIEW_VALUE,
+                getGerritBuildSuccessfulCodeReviewValue() != null ? getGerritBuildSuccessfulCodeReviewValue() : DEFAULT_GERRIT_BUILD_SUCCESSFUL_CODE_REVIEW_VALUE,
+                getGerritBuildFailedCodeReviewValue() != null ? getGerritBuildFailedCodeReviewValue() : DEFAULT_GERRIT_BUILD_FAILURE_CODE_REVIEW_VALUE,
+                getGerritBuildUnstableCodeReviewValue() != null ? getGerritBuildUnstableCodeReviewValue() : DEFAULT_GERRIT_BUILD_UNSTABLE_CODE_REVIEW_VALUE,
+                getGerritBuildNotBuiltCodeReviewValue() != null ? getGerritBuildNotBuiltCodeReviewValue() : DEFAULT_GERRIT_BUILD_NOT_BUILT_CODE_REVIEW_VALUE),
+            new VerdictCategory("Verified", "Verified",
+                getGerritBuildStartedVerifiedValue() != null ? getGerritBuildStartedVerifiedValue() : DEFAULT_GERRIT_BUILD_STARTED_VERIFIED_VALUE,
+                getGerritBuildSuccessfulVerifiedValue() != null ? getGerritBuildSuccessfulVerifiedValue() : DEFAULT_GERRIT_BUILD_SUCCESSFUL_VERIFIED_VALUE,
+                getGerritBuildFailedVerifiedValue() != null ? getGerritBuildFailedVerifiedValue() : DEFAULT_GERRIT_BUILD_FAILURE_VERIFIED_VALUE,
+                getGerritBuildUnstableVerifiedValue() != null ? getGerritBuildUnstableVerifiedValue() : DEFAULT_GERRIT_BUILD_UNSTABLE_VERIFIED_VALUE,
+                getGerritBuildNotBuiltVerifiedValue() != null ? getGerritBuildNotBuiltVerifiedValue() : DEFAULT_GERRIT_BUILD_NOT_BUILT_VERIFIED_VALUE)
+        };
+        for (VerdictCategory category : defaultCategories) {
+            if (!categories.contains(category)) {
+                categories.add(category);
+            }
+        }
+        List<VerdictCategory> categoriesToRemove = new ArrayList<>();
+        for (VerdictCategory cat : categories) {
+            if (cat.getVerdictValue().equalsIgnoreCase("CRVW") || cat.getVerdictValue().equalsIgnoreCase("VRIF"))
```

Comment:
in line 297 you create the `VerdictCategory` with the value `Code-Review` but here you look for `CRVW`. Shouldn't you look for `Code-Review`?

Reply:
The whole purpose behind this for statement is that somehow (couldn't figure out why) the labels "VRIF" && "CRVW" get automatically added in categories, so I removed them and migrated to Code-Review & Verified. Seems to work OK with previous configurations.